server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStore.java

```diff
@@ -190,6 +201,7 @@ public BucketName getDefaultBucketName() {
     public Mono<Void> deleteBucket(BucketName bucketName) {
         ObjectStorageBucketName resolvedBucketName = bucketNameResolver.resolve(bucketName);
         return Mono.<Void>fromRunnable(() -> blobStore.deleteContainer(resolvedBucketName.asString()))
+            .then(blobExistenceTester.truncateData()) // We truncate the all table here to avoid a full scan on it
```

Comment:
This Cassandra specific comment in a non cassandra class is just wrong.

Please remove it.

Maybe here we want to `deleteBucket` and the fact that we delete a bit more than the bucket is an implementation from the blobTester?

Reply:
There is a short timewindow between blob deletion & cache cleanup.

I suggest cleaning up the cache first.